### PR TITLE
Phantm link

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Links from variant and gene level to cancer dbs.

Screenshot 2020-11-29 at 12 30 02

Screenshot 2020-11-30 at 08 48 49

Screenshot 2020-11-29 at 13 30 10

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
Almost ready, but the cBioPortal variant link still seems a little off - perhaps we need to use the terser AA shorthand, so R273H instead of Arg273His.

**Comment by dnil** [comment]
Ok, ready for review!

**Comment by northwestwitch** [comment]
I can review

**Comment by northwestwitch** [comment]
I'm trying to test it locally, since I see you added the gene to the demo case. But the variant looks like this:

image

I think it's because TP53 doesn't get loaded by default in the demo.

But it does work after you run `scout --demo update genes`:

image

Is there an easy way to load TP53 as well whenever you setup the scout demo?

**Review by northwestwitch — APPROVED** [review]
It's working! Nice job!

I have some comments that are easy to address and one that I'm not really sure. The thing is since you are including TP53 in the VCF of the demo case it would be also nice to display all these nice links in the cancer demo case. But TP53 is not loaded so you won't see them unless you load all genes.. I don't know how complicated it would be to load it together with the other demo genes..

File: scout/server/links.py [hunk 1]
```diff
@@ -447,6 +479,32 @@ def ucsc_link(variant_obj, build=None):
     return url_template.format(this=variant_obj)
 
 
+def cbioportal(hgnc_symbol, protein_sequence_name):
```
Comment (northwestwitch) [inline scout/server/links.py:482]: This one could be tested

Comment (dnil) [reply scout/server/links.py:482]: Indeed! Will do!

File: scout/server/blueprints/genes/controllers.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def gene(store, hgnc_id):
                 transcript["position"] = "{this[chrom]}:{this[start]}-{this[end]}".format(
                     this=transcript
                 )
-                add_tx_links(transcript, build)
+                add_tx_links(transcript, build, record["hgnc_symbol"])
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/genes/controllers.py:40]: It fails 14 lines before? 😉

[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/genes/controllers.py:40]: Problem solved then!

File: scout/server/blueprints/variant/templates/variant/variant.html [hunk 3]
```diff
@@ -624,19 +624,29 @@ <h4 class="modal-title">Cancel Sanger sequencing</h4>
...
+              <a href="{{ gene.tp53_link }}" class="btn btn-outline-secondary" target="_blank">MutanTP53</a>
+            {% endif %}
+            {% if gene.iarctp53_link %}
+              <a href="{{ gene.iarctp53_link }}" class="btn btn-outline-secondary" target="_blank">IARC TP53</a>
+            {% endif %}
+            <a href="{{ gene.cbioportal_link }}" class="btn btn-outline-secondary" target="_blank">cBioPortal</a>
+            <a href="{{ gene.oncokb_link }}" class="btn btn-outline-secondary" target="_blank">OncoKB</a>
+          {% endif %}
+        </div>
+        <div class="btn-group-justified">
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant.html:638]: Sure, nice solution! One could also make everything a big form, but that would be harder to read. The sad part is that the MARRVEL button probably does not get much use. It is broken in production and no-one seems to have noticed. 🙄

[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/variant.html:638]: whoops

**Comment by dnil** [comment]
> Is there an easy way to load TP53 as well whenever you setup the scout demo?

I don't know about simple, but I agree that makes sense, and have added TP53 to the reduced resources. Its fine for a gene or two, but gets old and error prone pretty quickly so we should consider updating the scripts for this if it happens again..

File: scout/server/links.py (hunk 1 above)
Comment (dnil) [reply scout/server/links.py:497]: I settled with testing the function it calls. The rest should be very straightforward (famous last words).

File: scout/server/blueprints/variant/templates/variant/variant.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant.html]: I made an issue, and will leave it at this for now!

**Pull request merged**